content/ADR-239-avatar-assembly-changes.md

```diff
@@ -0,0 +1,197 @@
+To standardize this an ordered list of categories will be followed to calculate which wearables are hiding which.
+The proposed priority list is (from most important to least):
+- Skins
+- Top Body
+- Bottom Body
+- Shoes
+- Helmets
+- Masks
+- Hats
+- Top Head
+- Hair
+- Tiaras
+- Eyewear
+- Earrings
+
+All elements associated with the face, such as the mouth, eyes, and eyebrows, directly depend on the visualization of the head. The head is not a wearable itself, so it cannot create circular references, and it is hidden as soon as one or more wearables determine it.
+
+The following example illustrates the expected behavior: If a wearable of type "Mask" hides another wearable of type "Hat", and this in turn hides wearables of type "Mask", the wearable of type "Mask" will have visualization priority, hiding the Mask.
+
+
+### Hiding and replacing
+
+The current result of equipping an item that replaces or hides other wearables is unexpected and inconsistent. When a player navigates through their inventory items, as they equip, unequip, or hide other wearables from other categories without notice, they cannot reverse this behavior when equipping a different wearable.
+When a wearable unequips another wearable, the latter could be running a portable experience. By deactivating it, any progress or data in memory could be lost, ruining its expected behavior.
+The decision-making capacity of creators is overlapping with the freedom of choice of players. When a player purchases a wearable, thinking it will combine well with what they already have in their inventory, they may be unpleasantly surprised that these are not compatible. Information about which other wearables are replaced or hidden is not clearly indicated in the marketplace or backpack.
+There is no real weighty reason for a wearable to unequip another wearable, instead of hiding it to prevent clipping problems.
+
+On the base of that the replacing functionality for new wearables will be removed, pre-existing wearables will see the replacing list being merged with the hiding list.
+
+### Body fallback
+
+Previously, by unequipping the upper_body, the lower_body, the feet or the eyebrows, caused the system to fallback to default wearables (t-shirt, pants and shoes).
+This prevents, for example, players from going barefoot or without t-shirt.
+
+The system will extent the concept of Body by removing the default fallback wearables.
+These pieces, which can be called "Bare feet", "Naked Top" and "Naked Bottom", are not considered wearables, but rather parts of the base body geometry that are exclusively displayed when no wearables are equipped and there are no other wearables hiding the relevant category.
```

Comment:
A stronger algorithmic definition of these rules will be needed to implement the logic. What can we infer is only `name.isOneOf(["Bare feet", "Naked Top", "Naked Bottom"])` and that may not be enough